android/src/main/java/com/baseflow/permissionhandler/PermissionUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.baseflow.permissionhandler;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+import androidx.core.app.ActivityCompat;
+
+public class PermissionUtils {
+
+  @RequiresApi(api = Build.VERSION_CODES.M)
+   static boolean neverAskAgainSelected(final Activity activity, final String permission) {
+    final boolean prevShouldShowStatus = getRationaleDisplayStatus(activity, permission);
+    final boolean currShouldShowStatus = ActivityCompat.shouldShowRequestPermissionRationale(activity, permission);
+    return prevShouldShowStatus != currShouldShowStatus;
+  }
+
+   static void setShouldShowStatus(final Context context, final String permission) {
```

Comment:
This method name suggests we are storing the outcome of the `shouldShowRequestPermissionRationale` which is not the case since you are simply storing the fact that the permissions have been requested. I suggest you rename this method to something like (note that this line also starts with one extra space that shouldn't be there):
```suggestion
  static void setRequestedPermission(final Context context, final String permission) {
```